### DISCUSSION TITLE
post/naudio-wavestream-in-depth

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#69] [Announcements]
# post/naudio-wavestream-in-depth

Mark Heath's Development Blog

https://www.markheath.net/post/naudio-wavestream-in-depth

**Comment by wellumies** [comment]
I'm trying to write a game, that selects random 15s from a song. Fades into the song and fades out in 13-15s. I got the random part sorted with Skip, but how can I get the fade-in and fade-out and only play 15 seconds? There are no helper methods for that

**Reply by markheath** [reply]
There is a very basic `FadeInOutSampleProvider` in NAudio, which can be a bit tricky to use as you need to trigger the fade out at the right point, but hopefully will give you some ideas
